UnityProject/Assets/Scripts/Managers/CargoManager.cs

```diff
@@ -59,6 +59,23 @@ private void OnDisable()
 		SceneManager.activeSceneChanged -= OnRoundRestart;
 	}
 
+	/// <summary>
+	/// Checks for any Lifeforms (dead or alive) that might be aboard the shuttle
+	/// </summary>
+	/// <returns>true if a lifeform is found, false if none is found</returns>
+	bool LifeformChecker()
+	{
+		Transform ObjectHolder = CargoShuttle.Instance.SearchForObjectsOnShuttle();
+		for (int i = 0; i < ObjectHolder.childCount; i++)
+		{
+			if (ObjectHolder.GetChild(i).GetComponent<LivingHealthBehaviour>() != null)
```

Comment:
This is really expensive way to check if an object is a player. Did you try to check by layers?
<https://docs.unity3d.com/Manual/Layers.html> 
There is a Player layer.

Reply:
`if(((1<<gameObject.layer) & layers) == 0)` return `true`;
Where layers has Player layer. 
<https://docs.unity3d.com/ScriptReference/LayerMask.html>

Reply:
You can do ` LayerMask.GetMask("Player")`, but cache it before iterating in `for` to not call it everytime.

Reply:
Also you can iterate without using `for`
```
foreach (Transform child in transform)
{
  //child is your child transform
}
```